clients/src/main/java/org/apache/kafka/server/authorizer/Authorizer.java

```diff
@@ -114,6 +114,21 @@
      * This is an asynchronous API that enables the caller to avoid blocking during the update. Implementations of this
      * API can return completed futures using {@link java.util.concurrent.CompletableFuture#completedFuture(Object)}
      * to process the update synchronously on the request thread.
+     * <p>
+     * <b>Concurrent updates:</b>
+     * <ul>
+     *     <li>If ACLs are created using {@link #createAcls(AuthorizableRequestContext, List)} while a delete is in
+     *     progress, these ACLs may or may not be considered for deletion depending on the order of updates.
+     *     The returned {@link AclDeleteResult} indicates which ACLs were deleted.</li>
+     *     <li>If the provided filters use resource pattern type
+     *     {@link org.apache.kafka.common.resource.PatternType#MATCH} that needs to filter all resources to determine
+     *     matching ACLs, only ACLs that have already been propagated to the broker processing the ACL update will be
+     *     deleted. This may not include some ACLs that were persisted, but not yet propagated to all brokers. The
+     *     returned {@link AclDeleteResult} indicates which ACLs were deleted.</li>
+     *     <li>If the provided filters use other resource pattern types that perform a direct match, all matching ACLs
+     *     from previously completed {@link #createAcls(AuthorizableRequestContext, List)} )} are guaranteed
+     *     to be deleted.</li>
```

Comment:
@ijuma I added these to the Authorizer API (hence Admin API as well). But I wasn't sure if we should leave it to the implementation instead - i.e. implementations may provider stronger or weaker guarantees. If so, this doc would move to AclAuthorizer. What do you think?

Reply:
I agree,  maybe  we can move this  doc to `AclAuthorizer`.  we can also mention here to refer implementation docs for concurrent update guarantees.